src/components/tl/mlx5/mcast/tl_mlx5_mcast.h

```diff
@@ -633,4 +640,38 @@ ucc_status_t ucc_tl_mlx5_mcast_context_init(ucc_tl_mlx5_mcast_context_t *mcast_c
 
 
 ucc_status_t ucc_tl_mlx5_mcast_clean_ctx(ucc_tl_mlx5_mcast_coll_context_t *ctx);
+
+/* Mcast-specific logging wrapper macro that handles FORCE/TRY modes */
+#define tl_mlx5_mcast_log(_comm, _intended_level, _fmt, ...) \
+    do { \
+        if ((_comm)->context->mcast_enabled == UCC_TL_MLX5_MCAST_FORCE) { \
+            ucc_log_component((_intended_level), (_comm)->lib->log_component, (_fmt), ##__VA_ARGS__); \
+        } else if ((_comm)->context->mcast_enabled == UCC_TL_MLX5_MCAST_TRY && \
+                   (_comm)->lib->log_component.log_level <= UCC_LOG_LEVEL_DEBUG) { \
+            ucc_log_component(UCC_LOG_LEVEL_DEBUG, (_comm)->lib->log_component, (_fmt), ##__VA_ARGS__); \
+        } \
+    } while(0)
+
+/* Context-based logging wrapper for cases where we have ctx instead of comm */
+#define tl_mlx5_mcast_log_ctx(_ctx, _lib, _intended_level, _fmt, ...) \
+    do { \
+        if ((_ctx)->mcast_enabled == UCC_TL_MLX5_MCAST_FORCE) { \
+            ucc_log_component((_intended_level), (_lib)->log_component, (_fmt), ##__VA_ARGS__); \
+        } else if ((_ctx)->mcast_enabled == UCC_TL_MLX5_MCAST_TRY && \
+                   (_lib)->log_component.log_level <= UCC_LOG_LEVEL_DEBUG) { \
+            ucc_log_component(UCC_LOG_LEVEL_DEBUG, (_lib)->log_component, (_fmt), ##__VA_ARGS__); \
+        } \
+    } while(0)
+
+/* Context params-based logging wrapper for cases where we have ctx_params instead of ctx */
+#define tl_mlx5_mcast_log_ctx_params(_ctx_params, _lib, _intended_level, _fmt, ...) \
+    do { \
+        if ((_ctx_params)->mcast_enabled == UCC_TL_MLX5_MCAST_FORCE) { \
+            ucc_log_component((_intended_level), (_lib)->log_component, (_fmt), ##__VA_ARGS__); \
+        } else if ((_ctx_params)->mcast_enabled == UCC_TL_MLX5_MCAST_TRY && \
+                   (_lib)->log_component.log_level <= UCC_LOG_LEVEL_DEBUG) { \
+            ucc_log_component(UCC_LOG_LEVEL_DEBUG, (_lib)->log_component, (_fmt), ##__VA_ARGS__); \
+        } \
+    } while(0)
```

Comment:
maybe i missing something but if log_level < UCC_LOG_LEVEL_DEBUG then ucc_log_component(UCC_LOG_LEVEL_DEBUG, (_lib)->log_component, (_fmt), ##__VA_ARGS__); will not print anything. I think this logic will be more clear

```
        if (mcast_enabled == UCC_TL_MLX5_MCAST_TRY) {        ucc_log_component(UCC_LOG_LEVEL_DEBUG, (_lib)->log_component, (_fmt), ##__VA_ARGS__); }
       else {
ucc_log_component(_intended_level, (_lib)->log_component, (_fmt), ##__VA_ARGS__); }
```